v0/destinations/hs/transform.js

```diff
@@ -1,244 +1,609 @@
+    // Private Apps
+    response.headers = {
+      ...response.headers,
+      Authorization: `Bearer ${Config.accessToken}`
+    };
+  } else {
+    // use legacy API Key
+    response.params = { hapikey: Config.apiKey };
+  }
 
   return response;
-}
+};
 
-async function processTrack(message, destination, propertyMap) {
-  const parameters = {
-    _a: destination.Config.hubID,
-    _n: message.event
-  };
+/**
+ * using legacy API
+ * Reference:
+ * https://legacydocs.hubspot.com/docs/methods/contacts/create_contact
+ * https://legacydocs.hubspot.com/docs/methods/contacts/create_or_update
+ * @param {*} message
+ * @param {*} destination
+ * @param {*} propertyMap
+ * @returns
+ */
+const processLegacyIdentify = async (message, destination, propertyMap) => {
+  const { Config } = destination;
+  const traits = getFieldValueFromMessage(message, "traits");
+  const mappedToDestination = get(message, MappedToDestinationKey);
+  // if mappedToDestination is set true, then add externalId to traits
+  if (mappedToDestination) {
+    addExternalIdToTraits(message);
+  }
 
-  if (
-    message.properties &&
-    (message.properties.revenue || message.properties.value)
-  ) {
-    // eslint-disable-next-line dot-notation
-    parameters["_m"] = message.properties.revenue || message.properties.value;
+  if (!traits || !traits.email) {
+    throw new CustomError(
+      "[HS]:: Identify without email is not supported.",
+      400
+    );
   }
+
   const userProperties = await getTransformedJSON(
     message,
-    hSIdentifyConfigJson,
+    hsCommonConfigJson,
     destination,
     propertyMap
   );
 
-  return responseBuilderSimple(
-    { ...parameters, ...userProperties },
+  const payload = {
+    properties: formatPropertyValueForIdentify(userProperties)
+  };
+
+  const { email } = traits;
+  let endpoint;
+  if (email) {
+    endpoint = IDENTIFY_CREATE_UPDATE_CONTACT.replace(":contact_email", email);
+  } else {
+    endpoint = IDENTIFY_CREATE_NEW_CONTACT;
+  }
+
+  const response = defaultRequestConfig();
+  response.endpoint = endpoint;
+  response.method = defaultPostRequestConfig.requestMethod;
+  response.headers = {
+    "Content-Type": "application/json"
+  };
+  response.body.JSON = removeUndefinedAndNullValues(payload);
+
+  // choosing API Type
+  if (Config.authorizationType === "newPrivateAppApi") {
+    // Private Apps
+    response.headers = {
+      ...response.headers,
+      Authorization: `Bearer ${Config.accessToken}`
+    };
+  } else {
+    // use legacy API Key
+    response.params = { hapikey: Config.apiKey };
+  }
+
+  return response;
+};
+
+/**
+ * CRM Custom Objects
+ * Ref - https://developers.hubspot.com/docs/api/crm/crm-custom-objects
+ * Associations v3
+ * Ref - https://developers.hubspot.com/docs/api/crm/associations/v3
+ * @param {*} message
+ * @param {*} destination
+ * @param {*} propertyMap
+ */
+const processCRMCustomObjects = async (message, destination, traits) => {
```

Comment:
associations will be supported as our next task, lets remove this and store the code locally somewhere, so that work is lesser next time. HAVE NOT REVIEWED this one

Reply:
commented the function